pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -521,6 +481,7 @@ func (cb *ClusterBuilder) buildInboundClusterForPortOrUDS(clusterPort int, bind
 	return localCluster
 }
 
+// TODO remove this
 func (cb *ClusterBuilder) buildLocalityLbEndpoints(proxyView model.ProxyView, service *model.Service,
```

Comment:
generally agree that removing this code duplication is probably worth it, unless spreading the mutability of EDS that @howardjohn was talking about is both too difficult and completely necessary. because off the cuff id rather remove the mutability property for EDS and use snapshotting or something else since that's easier to reason about.

why is EDS handled differently? performance reasons? EDS being more dynamic?